lib/portage/dbapi/bintree.py

```diff
@@ -146,23 +148,33 @@ def aux_get(self, mycpv, wants, myrepo=None):
 		elif not self.bintree._remotepkgs or \
 			not self.bintree.isremote(mycpv):
 			try:
-				tbz2_path = self.bintree._pkg_paths[instance_key]
+				binpkg_path = self.bintree._pkg_paths[instance_key]
 			except KeyError:
 				raise KeyError(mycpv)
-			tbz2_path = os.path.join(self.bintree.pkgdir, tbz2_path)
+			binpkg_path = os.path.join(self.bintree.pkgdir, binpkg_path)
 			try:
-				st = os.lstat(tbz2_path)
+				st = os.lstat(binpkg_path)
 			except OSError:
 				raise KeyError(mycpv)
-			metadata_bytes = portage.xpak.tbz2(tbz2_path).get_data()
+			if binpkg_path.endswith(SUPPORTED_XPAK_EXTENSIONS):
+				metadata_bytes = portage.xpak.tbz2(binpkg_path).get_data()
+			elif binpkg_path.endswith(SUPPORTED_GPKG_EXTENSIONS):
+				metadata_bytes = portage.gpkg.gpkg(self.settings, mycpv,
+					binpkg_path).get_metadata()
+			else:
+				return
+
 			def getitem(k):
 				if k == "_mtime_":
 					return str(st[stat.ST_MTIME])
 				elif k == "SIZE":
 					return str(st.st_size)
-				v = metadata_bytes.get(_unicode_encode(k,
-					encoding=_encodings['repo.content'],
-					errors='backslashreplace'))
+				if binpkg_path.endswith(SUPPORTED_XPAK_EXTENSIONS):
+					v = metadata_bytes.get(_unicode_encode(k,
+						encoding=_encodings['repo.content'],
+						errors='backslashreplace'))
+				elif binpkg_path.endswith(SUPPORTED_GPKG_EXTENSIONS):
+					v = metadata_bytes.get(k)
```

Comment:
We can avoid having separate cases for xpak and gpkg here if we encode the keys for xpak earlier, like this:
```
if binpkg_path.endswith(SUPPORTED_XPAK_EXTENSIONS):
    metadata_bytes = portage.xpak.tbz2(binpkg_path).get_data()
    wants = [_unicode_encode(k, encoding=_encodings['repo.content'], errors='backslashreplace']
        for k in wants]
```